bin/extract_codeobject

```diff
@@ -0,0 +1,426 @@
+#!/usr/bin/perl
+use strict;
+use File::Copy;
+use File::Spec;
+use File::Basename;
+use File::Which;
+use Cwd 'realpath';
+use Getopt::Std;
+use List::Util qw(max);
+
+sub usage {
+  print("Usage: $0 [OPTION]... -i <input>\n");
+  print("Extract the device kernels from an hcc executable.\n\n");
+  print("-h                                \tshow this help message\n");
+  print("-i <input>                        \t[DEPRECATED] input file\n");
+  print("-E <kernel-name>[:<kernel-name>]  \tExtract code object by URI name. Multiple code object names may be specified using \':\' delimiter.\n");
+  print("-L                                \tList code object bundles, by entry name, and their embedded kernel code objects, by URI, available in source input file (-i option).\n");
+  print("-b                                \tList only bundle entries.\n");
+  print("-c <bundle-entry-name>            \tList only embedded code objects for specific code object bundle <bundle-entry-name>\n");
```

Comment:
Not sure on the meaning of these. What name is used to identify embedded bundled code objects? The section name they are in?

Does -c list code objects in embedded bundled code objects that match the provided name? Can wild cards be used? Is the main use of this to get the URI of a code object so it can be extracted using -u?

What file names are used for extracted code objects?

It just feels like this is really 2 tools. One that simply extracts code objects by a [list of] URI; and one that takes a [list of] host executabes and displays the embedded code objects. Perhaps this should be two suitably named tools with natural options? It that is done may want to consider an option to output to stdout so can pipe to other tools such as disassembly.

I think having some simple tools would fit better in the Unix style of composable tools.